set.mm

```diff
@@ -643671,6 +643722,980 @@ base set if and only if the neighborhoods (convergents) of every point
+       13-Aug-2023.) $)
+    expanduniss $p
+      |- ( U. A C_ B <-> A. x ( x e. A -> A. y ( y e. x -> y e. B ) ) ) $=
+  ( cuni wss cv wral wcel wi wal unissb dfss2 expandral bitri ) CEDFAGZDFZACHPC
+  IBGZPIRDIJBKZJAKACDLQSACBPDMNO $.
+  $}
+
+  ${
+    $d v u $.  $d v f $.  $d v i $.  $d u o $.  $d z w v $.  $d z v t $.
+    $d w v U $.  $d w o s $.
+    $( Express the predicate on ` y ` in ~ ismnu using only primitives.
+       (Contributed by Rohan Ridenour, 13-Aug-2023.) $)
+    ismnuprim $p
+      |- ( A. z e. U A. f ( ~P z C_ U /\ E. w e. U
+        ( ~P z C_ w /\ A. i e. z ( E. v e. U ( i e. v /\ v e. f )
+          -> E. u e. f ( i e. u /\ U. u C_ w ) ) )
+      )
+      <-> A. z ( z e. U -> A. f -. A. w ( w e. U -> -.
+        A. v -. ( ( A. t ( t e. v -> t e. z ) -> -. ( v e. U -> -. v e. w ) )
+          -> -. A. i ( i e. z -> ( v e. U -> ( i e. v -> ( v e. f ->
+            -. A. u ( u e. f -> ( i e. u ->
+              -. A. o ( o e. u -> A. s ( s e. o -> s e. w ) ) ) ) ) ) ) ) )
+      ) ) ) $=
+  ( cv wss wcel wa wrex wi wral wal wn bitri cpw wel cuni r19.42v w3a jcab pwss
+  19.26 albii anbi12i 3bitr4i ralcom4 19.23v 3anass df-rex bitr4i imbi1i ralbii
+  exbii bitr3i anass dfss2 imbi12i 3impexp biid expanduniss expandan expandrexn
+  wex df-an imbi2i expandral expandrex ) AKZUAZFLZVOBKZLZHKZCKZMZVTGKZMZNZCFOZH
+  DUBZDKZUCVQLZNZDWBOZPZHVNQZNZBFONZGRVQFMEKZVTMWOVNMPERZVTFMZCBUBZSPSZPZVSVNMW
+  QWAWCWGWBMWFIKZWGMJKZXAMXBVQMPJRPIRZSPZPDRSZPZPZPZPHRZSPSZCRZSPBRSZGRAFWNXLGW
+  NVPWMNZBFOXLVPWMBFUDXMXKBFXMVTVNLZWQWRNZPZWQWAWCUEZWJPZHVNQZNZCRZXKYAXPCRZXSC
+  RZNZXMXPXSCUHYDVPVRNZWLNXMYBYEYCWLXNWQPZXNWRPZNZCRYFCRZYGCRZNYBYEYFYGCUHXPYHC
+  XNWQWRUFUIVPYIVRYJCVNFUGCVNVQUGUJUKYCXRCRZHVNQWLXRHCVNULYKWKHVNYKXQCVIZWJPWKX
+  QWJCUMYLWEWJYLWQWDNZCVIWEXQYMCWQWAWCUNUSWDCFUOUPUQTURUTUJVPVRWLVATTXTXJCXPWTX
+  SXIXNWPXOWSEVTVNVBWQWRVJVCXRXHHVNXRWQWAWCWJPZPZPXHWQWAWCWJVDYOXGWQYNXFWAWJXEW
+  CWIXDDWBWFWFWHXCWFVEIJWGVQVFVGVHVKVKVKTVLVGUIUTVMUTUIVL $.
+  $}
+
+  ${
+    $d u o $.  $d z v t $.  $d w o s $.  $d y z w v u f i $.
+    $( Express "every set is contained in a Grothendieck universe" using only
+       primitives.  The right side uses only 123 symbols (not counting the
+       outer universal quantifier, which would be unnecessary if it were an
+       axiom), which is significantly less than the previous record of 163
+       esablished by ~ grothprim (which uses some defined symbols, and requires
+       229 symbols if expanded to primitives).  (Contributed by Rohan Ridenour,
+       13-Aug-2023.) $)
+    rr-grothprim $p |- ( A. x E. y e. Univ x e. y <->
+      A. x -. A. y ( x e. y -> -. A. z ( z e. y
+        -> A. f -. A. w ( w e. y -> -. A. v -.
+          ( ( A. t ( t e. v -> t e. z ) -> -. ( v e. y -> -. v e. w ) )
+            -> -. A. i ( i e. z -> ( v e. y -> ( i e. v -> ( v e. f ->
+              -. A. u ( u e. f -> ( i e. u ->
+                -. A. o ( o e. u -> A. s ( s e. o -> s e. w ) ) ) ) ) ) ) ) ) )
+      ) ) ) $=
```

Comment:
Would the RHS be nicer if you allowed E., \/, /\ , C_ as well as restricted quantifiers ?

Reply:
I guess that would look like this, simply applying `~ ismnu`:
```
A. x E. y ( x e. y /\ A. z e. y A. f ( ~P z C_ y /\ E. w e. y
        ( ~P z C_ w /\ A. i e. z
          ( E. v e. y ( i e. v /\ v e. f )
            -> E. u e. f ( i e. u /\ U. u C_ w ) ) ) ) ) ) )
```
(worth adding as a theorem, in my opinion)

Reply:
`A. z e. y A. f ( ~P z C_ y /\ ...`
-->
`A. z e. y ( ~P z C_ y /\ A. f ...`
would probably be a bit simpler.

Reply:
Added as ~ rr-groth .